### PR TITLE
feat(driver): hoist magento into subpackage

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Moves `driver/magento` into its own package.
- Changes auth magento driver to use the new package.

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/package.json [hunk 1]
```diff
@@ -25,10 +25,12 @@
     "@angular/common": "0.0.0-PLACEHOLDER",
     "@angular/core": "0.0.0-PLACEHOLDER",
     "@daffodil/core": "0.0.0-PLACEHOLDER",
+    "@daffodil/driver/magento": "0.0.0-PLACEHOLDER",
```
Comment (damienwebdev) [inline libs/auth/package.json:28]: We don't publish sub packages independently of the main package, so the dep here is simply `@daffodil/driver`.

File: libs/auth/package.json [hunk 2]
```diff
@@ -25,10 +25,12 @@
...
     "@ngrx/store": "0.0.0-PLACEHOLDER",
     "rxjs": "0.0.0-PLACEHOLDER"
   },
   "devDependencies": {
-    "@daffodil/core": "0.0.0-PLACEHOLDER"
+    "@daffodil/core": "0.0.0-PLACEHOLDER",
+    "@daffodil/driver/magento": "0.0.0-PLACEHOLDER"
```
Comment (damienwebdev) [inline libs/auth/package.json:34]: We don't publish sub packages independently of the main package, so the dep here is simply `@daffodil/driver`.

File: libs/auth/package.json (hunk 1 above)
Comment (damienwebdev) [reply libs/auth/package.json:28]: This dep is also an `optionalDependency`.

Comment (griest024) [reply libs/auth/package.json]: fixed

File: libs/auth/package.json (hunk 2 above)
Comment (griest024) [reply libs/auth/package.json]: fixed

File: libs/auth/package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "@angular/common": "0.0.0-PLACEHOLDER",
     "@angular/core": "0.0.0-PLACEHOLDER",
     "@daffodil/core": "0.0.0-PLACEHOLDER",
+    "@daffodil/driver": "0.0.0-PLACEHOLDER",
```
Comment (damienwebdev) [inline libs/auth/package.json:28]: Remove, as this is not a peer dep.

Comment (griest024) [reply libs/auth/package.json]: changed

**Pull request merged**